compiler/rustc_target/src/abi/call/mod.rs

```diff
@@ -605,7 +605,7 @@ impl<'a, Ty> FnAbi<'a, Ty> {
             "nvptx64" => nvptx64::compute_abi_info(self),
             "hexagon" => hexagon::compute_abi_info(self),
             "riscv32" | "riscv64" => riscv::compute_abi_info(cx, self),
-            "wasm32" if cx.target_spec().os != "emscripten" => {
+            "wasm32" if cx.target_spec().os == "bindgen" => {
```

Comment:
Does it still make sense to change this for WASI and leave `unknown` alone in the short term? While wasm_bindgen may change in the future so that unknown can be changed, it would be nice to go ahead and change the ABI for *some* target so that its possible to build C+WASM code that does struct passing by-value even if it cannot be interoperated with wasm_bindgen immediately.

Reply:
Yeah that sounds reasonable to me!